core/shared/src/main/scala/cats/parse/Parser.scala

```diff
@@ -189,6 +189,24 @@ sealed abstract class Parser[+A] {
   def map[B](fn: A => B): Parser[B] =
     Parser.map(this)(fn)
 
+  /** Transform parsed values using the given function, or fail on None
+    *
+    * When the function return None, this parser fails
+    * This is implemented with select, which makes it more efficient
+    * than using flatMap
+    */
+  def mapOrFail[B](fn: A => Option[B]): Parser[B] = {
```

Comment:
This looks like `FunctorFilter#mapFilter`.  Is that a better name?

Reply:
I was thinking the same. Could we add a `def collect[B](fn: PartialFunction[A, B]): Parser[B]` as well?

Reply:
great review folks, I added a FunctorFilter implementation and used these names.

Please take a look. I'd like to publish 0.2.0 after merging this unless you disagree.